Updated the SparkHistoryServer docker shell to make it user friendly

krishnadasmallya opened the pull request:
Was looking for a quick and easy way to run SparkHistoryServer locally and found this. Made some updates to the script for Spark History Server Docker to make it more descriptive and easy to configure.

vara-bonthu commented on docker/README.md:
```diff
@@ -13,17 +14,19 @@
-git clone https://github.com/Hyper-Mesh/spark-history-server.git
-cd docker
-docker build -t <docker-user-id>/spark-web-ui:latest . 
+git clone https://github.com/krishnadasmallya/spark-history-server.git
```
This is pointing to your personal repo. Can this be changed to git clone https://github.com/Hyper-Mesh/spark-history-server.git?

vara-bonthu commented on docker/README.md:
````diff
@@ -13,17 +14,19 @@
 ```
 
-### Run Spark History Server as a local docker contianer
+### Run Spark History Server as a local docker container
````
Nice

vara-bonthu commented on docker/launch_spark_history_server_locally.sh:
```diff
@@ -1,20 +1,92 @@
+	S3_BUCKET=$1
+	S3_BUCKET_PREFIX=$2
+	LOG_DIR="s3a://$S3_BUCKET/$S3_BUCKET_PREFIX"
+	DOCKER_IMAGE="$USER/spark-web-ui:latest"
```
Make this as variable as well. Because $USER has to be a input. `DOCKER_IMAGE="$USER/spark-web-ui:latest"`

vara-bonthu commented on docker/launch_spark_history_server_locally.sh:
```diff
@@ -1,20 +1,92 @@
+	LOG_DIR="s3a://$S3_BUCKET/$S3_BUCKET_PREFIX"
+	DOCKER_IMAGE="$USER/spark-web-ui:latest"
 
-AWS_ACCESS_KEY_ID=""
-AWS_SECRET_ACCESS_KEY=""
-AWS_SESSION_TOKEN=""
```
We need these as INPUT variables as well so that users can just change the values of variables in the Shell script without having to go through the rest of the code to modify.

krishnadasmallya replied:
Removed this from the script based on my understanding that most AuthN solution generally gives export commands for the users. 

I may have missed to update the README with this requirement. Having them copy the output then set them in the script may induce human errors. I was thinking its better if they can use it on the terminal or session they are in before invoking this script.

Having this in the script not only means that the user has to update the script each time for authentication but also means they may have to copy parts of the output from AuthN command they have for each of these three. 

Rather than asking the Customer to export these variables and having an error thrown if found empty would be better, do you agree? Intention here is that users do not have to edit the script and can run it using just updated arguments. Having these long values as arguments will be a pain to manage as well.

vara-bonthu commented on docker/launch_spark_history_server_locally.sh:
```diff
@@ -1,20 +1,92 @@
-LOG_DIR="s3a://<bucket_name>/<spark-event-logs-prefix>/"
-ENDPOINT="s3.<region>.amazonaws.com"
+CLASS="org.apache.spark.deploy.history.HistoryServer"
+AWS_REGION="${4:-us-east-1}"
```
Should this be as `$3`?  Like AWS_REGION="${3:-us-east-1}"

krishnadasmallya replied:
No, as per the command its the 4th argument after the action(start/stop) bucket and prefix.

vara-bonthu commented on docker/launch_spark_history_server_locally.sh:
```diff
@@ -1,20 +1,92 @@
+CLASS="org.apache.spark.deploy.history.HistoryServer"
+AWS_REGION="${4:-us-east-1}"
+ENDPOINT="s3.$AWS_REGION.amazonaws.com"
+CONTAINER_NAME="${5:-spark-history-server}"
```
Should this be as `$4`?  Like CONTAINER_NAME="${4:-spark-history-server}"

krishnadasmallya replied:
No, as per the command its the 5th argument after the action(start/stop), bucket, prefix and aws region.

vara-bonthu reviewed (changes requested):
Thanks @krishnadasmallya for this improvement 👍🏼  This is looks great. I have few comments on the PR

krishnadasmallya commented:
@vara-bonthu  Have made some major changes to use dynamic parsing of optional arguments as with these many arguments we cannot rely on order alone as you may want to pass 5th argument with 4 as default etc.

Hopefully this version addresses all your suggestions/comments, plaase review and let me know

vara-bonthu reviewed (approved):
Thanks @krishnadasmallya 👍 LGTM 🚀
